api/provider/aws/aws_test/load_balancer_create_test.go

```diff
@@ -0,0 +1,335 @@
+			Interval:           30,
+			Timeout:            5,
+			HealthyThreshold:   2,
+			UnhealthyThreshold: 2,
+		},
+	}
+
+	readSGHelper(mockAWS, "l0-test-env_id-env", "env_sg")
+
+	createSGHelper(t, mockAWS, "l0-test-lb_id-lb", "vpc_id")
+	readSGHelper(mockAWS, "l0-test-lb_id-lb", "lb_sg")
+
+	ingressInput := &ec2.AuthorizeSecurityGroupIngressInput{}
+	ingressInput.SetGroupId("lb_sg")
+	ingressInput.SetCidrIp("0.0.0.0/0")
+	ingressInput.SetIpProtocol("TCP")
+	ingressInput.SetFromPort(int64(80))
+	ingressInput.SetToPort(int64(80))
+
+	mockAWS.EC2.EXPECT().
+		AuthorizeSecurityGroupIngress(ingressInput).
+		Return(&ec2.AuthorizeSecurityGroupIngressOutput{}, nil)
+
+	iamRoleInput := &iam.CreateRoleInput{}
+	iamRoleInput.SetRoleName("l0-test-lb_id-lb")
+	iamRoleInput.SetAssumeRolePolicyDocument(provider.DEFAULT_ASSUME_ROLE_POLICY)
+
+	mockAWS.IAM.EXPECT().
+		CreateRole(iamRoleInput).
+		Return(&iam.CreateRoleOutput{}, nil)
+
+	renderedLBRolePolicy, err := provider.RenderLoadBalancerRolePolicy(
+		"region",
+		"123456789012",
+		"l0-test-lb_id",
+		provider.DEFAULT_LB_ROLE_POLICY_TEMPLATE)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	putIAMPolicyInput := &iam.PutRolePolicyInput{}
+	putIAMPolicyInput.SetPolicyName("l0-test-lb_id-lb")
+	putIAMPolicyInput.SetRoleName("l0-test-lb_id-lb")
+	putIAMPolicyInput.SetPolicyDocument(renderedLBRolePolicy)
+
+	mockAWS.IAM.EXPECT().
+		PutRolePolicy(putIAMPolicyInput)
+
+	listeners := make([]*elb.Listener, len(req.Ports))
+	listener := &elb.Listener{}
+	listener.SetProtocol("tcp")
+	listener.SetLoadBalancerPort(80)
+	listener.SetInstancePort(80)
+	listener.SetInstanceProtocol("tcp")
+	listeners[0] = listener
+
+	createLoadBalancerInput := &elb.CreateLoadBalancerInput{}
+	createLoadBalancerInput.SetLoadBalancerName("l0-test-lb_id")
+	createLoadBalancerInput.SetScheme("internet-facing")
+	createLoadBalancerInput.SetSecurityGroups([]*string{aws.String("env_sg"), aws.String("lb_sg")})
+	createLoadBalancerInput.SetSubnets([]*string{aws.String("pub1"), aws.String("pub2")})
+	createLoadBalancerInput.SetListeners(listeners)
+
+	mockAWS.ELB.EXPECT().
+		CreateLoadBalancer(createLoadBalancerInput).
+		Return(&elb.CreateLoadBalancerOutput{}, nil)
+
+	healthCheck := &elb.HealthCheck{}
+	healthCheck.SetTarget("TCP:80")
+	healthCheck.SetInterval(int64(30))
+	healthCheck.SetTimeout(int64(5))
+	healthCheck.SetHealthyThreshold(int64(2))
+	healthCheck.SetUnhealthyThreshold(int64(2))
+
+	configureHealthCheckInput := &elb.ConfigureHealthCheckInput{}
+	configureHealthCheckInput.SetLoadBalancerName("l0-test-lb_id")
+	configureHealthCheckInput.SetHealthCheck(healthCheck)
+
+	mockAWS.ELB.EXPECT().
+		ConfigureHealthCheck(configureHealthCheckInput).
+		Return(&elb.ConfigureHealthCheckOutput{}, nil)
+
+	target := provider.NewLoadBalancerProvider(mockAWS.Client(), tagStore, mockConfig)
+	result, err := target.Create(req)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, "lb_id", result)
+
+	expectedTags := models.Tags{
```

Comment:
can remove - handled in other unit test

Reply:
:white_check_mark:

Reply:
Should I remove it from here, or remove it from the Defaults test? For now, I've opted to remove it from the Defaults one.

Reply:
Remove it from defaults - the scope of that test does not include tags

Reply:
:white_check_mark: